src/main/java/org/sonarsource/sonarlint/ls/IssuesCache.java

```diff
@@ -102,6 +105,32 @@ public void removeIssueWithServerKey(String fileUriStr, String key) {
     }
   }
 
+  public Optional<Map.Entry<String, VersionedIssue>> findIssueWithId(String fileUriStr, String key) {
+    var fileUri = URI.create(fileUriStr);
+    var issues = issuesPerIdPerFileURI.get(fileUri);
+    if (issues != null) {
+      return issues.entrySet()
+        .stream()
+        .filter(issueEntry -> isDelegatingIssueWithKey(key, issueEntry))
+        .findFirst();
+    }
+    return Optional.empty();
+  }
+
+  public void updateIssueStatus(String fileUriStr, String key, HotspotStatus newStatus) {
```

Comment:
I think the name `key` is confusing here, as it actually refers to a `serverIssueKey`.

Reply:
Addressed in [c411b03](https://github.com/SonarSource/sonarlint-language-server/pull/252/commits/c411b03b7bc01dc3802ed6f65a0d8640a183059c)